### PR TITLE
build: support building with clang on ubuntu

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables building tachyon with clang on ubuntu 22.04

### CODE REVIEW

**Comment by chokobole** [comment]
Since a lint error has occurred in `tachyon/base/threading/platform_thread_linux.cc`, which belongs to Chromium, to keep the differences minimal, I kindly request that you ignore this :pray:

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: docs/how_to_use/how_to_build.md [hunk 1]
```diff
@@ -232,6 +232,28 @@ export CXX=/usr/bin/g++-10
 export GCC_HOST_COMPILER_PATH=/usr/bin/gcc-10
 ```
 
+### Building with Clang on Ubuntu 22.04
+
+To utilize features like [always_inline](https://releases.llvm.org/15.0.0/tools/clang/docs/AttributeReference.html#always-inline-force-inline),
+you'll need clang++ version 15 or higher. By default, Ubuntu 22.04 installs `clang-14` via `apt install clang`,
```
Comment (ashjeong) [inline docs/how_to_use/how_to_build.md:238]:
```suggestion
you'll need clang++ at version 15 or higher. By default, Ubuntu 22.04 installs `clang-14` via `apt install clang`,

File: docs/how_to_use/how_to_build.md [hunk 2]
```diff
@@ -232,6 +232,28 @@ export CXX=/usr/bin/g++-10
...
+sudo apt install clang-15 libomp5-15 libomp-15-dev
+```
+
+Next, update `CC` and `CXX` to point to the newly installed `clang-15` and `clang++-15`:
+And then make sure `CC` and `CXX` are updated to `clang-15` and `clang++-15`.
+
+```shell
+export CC=/usr/bin/clang-15
+export CXX=/usr/bin/clang++-15
+```
```
Comment (ashjeong) [inline docs/how_to_use/how_to_build.md:253]:
```suggestion
Next, update `CC` and `CXX` to point to the newly installed `clang-15` and `clang++-15`:

```shell
export CC=/usr/bin/clang-15
export CXX=/usr/bin/clang++-15
__ <- @ashjeong: ignore this - formatting is off

Make sure `CC` and `CXX` are properly updated to `clang-15` and `clang++-15`.
```

File: docs/how_to_use/how_to_build.md [hunk 3]
```diff
@@ -232,6 +232,28 @@ export CXX=/usr/bin/g++-10
 export GCC_HOST_COMPILER_PATH=/usr/bin/gcc-10
 ```
 
+### Building with Clang on Ubuntu 22.04
+
+To utilize features like [always_inline](https://releases.llvm.org/15.0.0/tools/clang/docs/AttributeReference.html#always-inline-force-inline),
+you'll need clang++ version 15 or higher. By default, Ubuntu 22.04 installs `clang-14` via `apt install clang`,
+so you'll need to manually install a newer version of clang and update `openmp` accordingly.
+
+First, install clang-15 and the appropriate OpenMP libraries:
```
Comment (ashjeong) [inline docs/how_to_use/how_to_build.md:241]:
```suggestion
First, install `clang-15` and the appropriate OpenMP libraries:

**Review by ashjeong — CHANGES_REQUESTED** [review]
https://github.com/kroma-network/tachyon/pull/414/commits/b30a8f6a3dff6d11989e7fa96fce01c8b958bb5a
Please change the commit title to a `constexpr` (with `) to match the commit before

https://github.com/kroma-network/tachyon/pull/414/commits/bd15530fbd82deccfc327a1e32ac05afaf637331
Please change the commit message to this:
When linking rust tests with clang, clang **detects and tries to use the system** installed intel openmp **even when compiling** without the option `--//:has_intel_openmp`,.

File: tachyon/base/threading/platform_thread_linux.cc [hunk 4]
```diff
@@ -134,11 +134,11 @@ long sched_setattr(pid_t pid,
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
 #if !BUILDFLAG(IS_NACL)
+/*
+TODO(chokobole):
 const char kCgroupDirectory[] =
     "/sys/fs/cgroup";
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/base/threading/platform_thread_linux.cc:140]: I should change the commit title!

**Review by Insun35 — APPROVED** [review]
LGTM

File: tachyon/base/threading/platform_thread_linux.cc [hunk 5]
```diff
@@ -134,11 +134,11 @@ long sched_setattr(pid_t pid,
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
 #if !BUILDFLAG(IS_NACL)
+/*
+TODO(chokobole):
```
[the remark this answers is not in the document]
Comment (ashjeong) [reply tachyon/base/threading/platform_thread_linux.cc:138]:
```
const char kCgroupDirectory[] =
     "/sys/fs/cgroup";
```
in the lines after is being put into comments and is the variable that is effectively being removed

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/414/commits/41d8c41ead42fbf626f336b01f7faa25032620b2
you've missed correcting the word "sytem" to "system"

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**